rbmc-cfam-daemon/test/local_cfam_test.cpp

```diff
@@ -0,0 +1,183 @@
+// SPDX-License-Identifier: Apache-2.0
+#include "local_cfam.hpp"
+#include "mock_sysfs.hpp"
+
+#include <gtest/gtest.h>
+
+using ::testing::Return;
+
+TEST(LocalCFAMTest, GetMaxValueTest)
+{
+    EXPECT_EQ(local_cfam_util::getMaxValue(4), 0xF);
+    EXPECT_EQ(local_cfam_util::getMaxValue(8), 0xFF);
+    EXPECT_EQ(local_cfam_util::getMaxValue(9), 0x1FF);
+    EXPECT_EQ(local_cfam_util::getMaxValue(24), 0x00FFFFFF);
+}
+
+TEST(LocalCFAMTest, TestWriteApiVersion)
+{
+    using Expected = std::expected<uint32_t, int>;
+    MockSysFS sysfs;
+    LocalCFAM cfam{0, sysfs};
+    std::filesystem::path reg{
+        "/sys/class/fsi-master/fsi0/slave@00:00/scratch1"};
+
+    EXPECT_CALL(sysfs, read(reg)).WillOnce(Return(Expected(0)));
```

Comment:
I have a general question regarding the TC expectations.

We are writing to a fixed location (cfam filesystem), but we're not currently checking whether the write operation returns as expected, correct?

Should we instead write to a temporary (mock) file and verify by reading it back after the write is completed to ensure the data was written properly?

Reply:
Good question.
The SysFS class handles reading and writing the files, so there is a sysfs test that deals with the low level file writes that does what you mention.  That leaves this code here easier to test.


https://github.com/ibm-openbmc/openpower-proc-control/blob/1110/rbmc-cfam-daemon/test/sysfs_test.cpp